client/my-sites/checkout/src/lib/get-plan-features.ts

```diff
@@ -42,17 +42,21 @@ export default function getPlanFeatures(
 			return [];
 		}
 
-		const featureList: Feature[] = planObject?.getCheckoutFeatures?.() || [];
+		const features: Feature[] = planObject?.getCheckoutFeatures?.() || [];
 		const annualPlanOnlyFeatures = planObject?.getAnnualPlansOnlyFeatures?.() || [];
 
-		return (
-			featureList
-				// Exclude annual plan only features if the current plan is a monthly plan
-				?.filter( ( feature ) => ! isMonthlyPlan || ! annualPlanOnlyFeatures.includes( feature ) )
-				// Show the free domain feature if `showFreeDomainFeature` is true
-				?.filter( ( feature ) => feature !== FEATURE_FREE_DOMAIN || showFreeDomainFeature )
-				?.map( ( feature ) => String( FEATURES_LIST[ feature ].getTitle() ) )
-		);
+		const featureList = features
+			// Exclude annual plan only features if the current plan is a monthly plan
+			.filter( ( feature ) => ! isMonthlyPlan || ! annualPlanOnlyFeatures.includes( feature ) )
+			// Show the free domain feature if `showFreeDomainFeature` is true
+			.filter( ( feature ) => feature !== FEATURE_CUSTOM_DOMAIN || showFreeDomainFeature )
+			.map( ( feature ) => String( FEATURES_LIST[ feature ].getTitle() ) );
+
+		// If the new feature list is available, return it.
+		// Else fallback to the previous list of features.
+		if ( featureList.length ) {
+			return featureList;
```

Comment:
Hey is the inline comment here correct? Should we be returning `features` on L58?